scipy/optimize/_linprog_util.py

```diff
@@ -1086,6 +1086,79 @@ def vstack(blocks):
     return A, b, c, c0, x0
 
 
+def _round_to_power_of_two(x):
+    """
+    Round elements of the array to the nearest power of two.
+    """
+    return 2**np.round(np.log2(x))
+
+def _autoscale(A, b, c, x0):
+    """
+    Scales the problem according to ___
+    """
+    m, n = A.shape
+
+#    with warnings.catch_warnings():
+#    warnings.simplefilter("ignore")
+    C = 1
+    R = 1
+
+    if A.size > 0:
+        if sps.issparse(A):
+            R = np.max(np.abs(A), axis=1).toarray().flatten()
+            R[R == 0] = 1
+            R = _round_to_power_of_two(R)
+            b = b/R
+            R = sps.diags(1/R)
+            A = R*A
+
+            C = np.max(np.abs(A), axis=0).toarray().flatten()
+            C[C == 0] = 1
+            C = _round_to_power_of_two(C)
+            c = c/C
+            C = sps.diags(1/C)
+            A = A*C
+
+        else:
+            R = np.max(np.abs(A), axis=1)  # problematic with sparse arrays
```

Comment:
I think the comment "problematic with sparse arrays" was that the operation returns a `matrix` instead of an array, and that was problematic later. It's not a bug; just an annoyance.

The comment below "no good when array is sparse" was that I'd get a warning doing:
```
R[R == 0] = 1 
```
when `R` is sparse, which makes sense. But I'm not really concerned with the speed of this operation because `R` has only one dimension and this only happens once, so for sparse matrices I just convert to dense.

I've removed the comments.